torchaudio/csrc/build_rir.cpp

```diff
@@ -0,0 +1,187 @@
+  torch::Tensor new_bands = torch::zeros({n, 2}, centers.dtype());
+  new_bands.requires_grad_(true);
+  scalar_t* newband_data = new_bands.data_ptr<scalar_t>();
+  const scalar_t* centers_data = centers.data_ptr<scalar_t>();
+  at::parallel_for(0, n, 0, [&](int64_t start, int64_t end) {
+    for (int64_t i = start; i < end; i++) {
+      if (i == 0) {
+        newband_data[i * 2] = centers_data[0] / 2;
+        newband_data[i * 2 + 1] = centers_data[1];
+      } else if (i == n - 1) {
+        newband_data[i * 2] = centers_data[n - 2];
+        newband_data[i * 2 + 1] = sample_rate / 2;
+      } else {
+        newband_data[i * 2] = centers_data[i - 1];
+        newband_data[i * 2 + 1] = centers_data[i + 1];
+      }
+    }
+  });
+  auto n_freq = n_fft / 2 + 1;
+  torch::Tensor freq_resp = torch::zeros({n_freq, n}, centers.dtype());
+  torch::Tensor freq =
+      torch::arange(n_freq, centers.dtype()) / n_fft * sample_rate;
+  const scalar_t* freq_data = freq.data_ptr<scalar_t>();
+  scalar_t* freqreq_data = freq_resp.data_ptr<scalar_t>();
+
+  at::parallel_for(0, n, 0, [&](int64_t start, int64_t end) {
+    for (auto i = start; i < end; i++) {
+      for (auto j = 0; j < n_freq; j++) {
+        if (freq_data[j] >= newband_data[i * 2] &&
+            freq_data[j] < centers_data[i]) {
+          freqreq_data[j * n + i] =
+              0.5 * (1 + cos(2 * M_PI * freq_data[j] / centers_data[i]));
+        }
+        if (i != n - 1 && freq_data[j] >= centers_data[i] &&
+            freq_data[j] < newband_data[i * 2 + 1]) {
+          freqreq_data[j * n + i] = 0.5 *
+              (1 - cos(2 * M_PI * freq_data[j] / newband_data[i * 2 + 1]));
+        }
+        if (i == n - 1 && centers_data[i] <= freq_data[j]) {
+          freqreq_data[j * n + i] = 1.0;
+        }
+      }
+    }
+  });
+  filters = torch::fft::fftshift(torch::fft::irfft(freq_resp, n_fft, 0), 0);
+  filters = filters.index({Slice(1)}).transpose(0, 1);
+}
+
+/**
+ * @brief Create the band-pass filters for the octave bands.
+ *
+ * @param centers The Tensor that stores the center frequencies of octave bands.
+ * Tensor with dimension `(num_band,)`.
+ * @param sample_rate The sample_rate of simulated room impulse response signal.
+ * @param n_fft The window size of FFT.
+ * @return torch::Tensor The output band-pass filter. Tensor with dimensions
+ * `(num_band, n_fft - 1)`.
+ */
+torch::Tensor make_filter(
+    torch::Tensor centers,
+    double sample_rate,
+    int64_t n_fft) {
+  torch::Tensor filters;
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+      centers.scalar_type(), "make_filter", [&] {
+        make_filter_impl<scalar_t>(centers, sample_rate, n_fft, filters);
+      });
+  return filters;
+}
+
+TORCH_LIBRARY_FRAGMENT(torchaudio, m) {
+  m.def(
+      "torchaudio::build_rir(Tensor irs, Tensor delay_i, int rir_length) -> Tensor",
+      &torchaudio::rir::build_rir);
+  m.def(
+      "torchaudio::make_filter(Tensor centers, float sample_rate, int n_fft) -> Tensor",
```

Comment:
The function name is very generic to be in torchaudio namespace. Can you make it more descriptive / specific?